storagenode/piecestore/endpoint.go

```diff
@@ -315,6 +338,24 @@ func (endpoint *Endpoint) Upload(stream pb.DRPCPiecestore_UploadStream) (err err
 	defer commitOrderToStore(ctx, &largestOrder)
 
 	for {
+		// Measure upload rate (bytes per second)
+		dt := time.Since(startTime)
+
+		// Skip flagging unusually slow connections in the first moments of uploading.
+		if dt > endpoint.config.DelayUntilSlowConnectionFlagged {
+			// currentUploadSize indicates the size of uploaded data until this moment.
```

Comment:
As the check is performed only on upload that maintains open connection longer than `DelayUntilSlowConnectionFlagged`, it enables a possibility to bypass small upload, if user decides to do so.